### DISCUSSION TITLE
Problem with ResourceIdResolver

### DISCUSSION THREAD

**Discussion by andyrozman** [awspring/spring-cloud-aws#311] [Q&A]
Hi !
So I am upgrading project from boot 2.2 to boot 2.6 (and to cloud 2021.0.1 and of course aws cloud 2.4.1).

In my project we have defined our own SqsConfiguration, but when I try to run new micorservice after I did upgrade, I get following error
Parameter 1 of method myMessagingTemplate in com.b.b.d.a.sqs.queue.assuerer.SqsQueueAssurerConfiguration required a bean of type 'io.awspring.cloud.core.env.ResourceIdResolver' that could not be found.

When I was testing on friday I didn't have the same problem, but I didn't use bootstrap dependency then (spring-cloud-starter-bootstrap).

Which beans do implement ResourceIdResolver, that I could use... I was considering doing something like:

@Bean
public ResourceIdResolver getResolver() {
// DON'T KNOW WHAT TO USE HERE
}

What would you recommend as solution? 

So far I could find only one class that implements that interface (StackResourceRegistryDetectingResourceIdResolver). Are there more classes that implemnt this? I tried looking in sources, but my Intellij is not cooperating today..

**Comment by maciejwalkowiak** [comment]
Unless you use CloudFormation integration you should not need `ResourceIdResolver`. Perhaps you can change `SqsQueueAssurerConfiguration` to not use it at all? Hard to guide without knowing exactly how this configuration class looks like

**Reply by andyrozman** [reply]
This is how class looks now (I added that method already)

{code}
@Configuration
@EnableAspectJAutoProxy
public class SqsQueueAssurerConfiguration {

    @Bean
    public SqsQueueBuilder sqsQueueBuilder() {
        return new SqsQueueBuilder();
    }

    @Bean
    public SqsQueueSender sqsQueueSender() {
        return new SqsQueueSender();
    }

    @Bean
    public SqsMethodInvocationProcessor sqsMethodInvocationProcessor() {

        return new SqsMethodInvocationProcessor();
    }

    @Bean
    @Primary
    public QueueMessagingTemplate myMessagingTemplate(AmazonSQSAsync amazonSqs, ResourceIdResolver resolver) {
        ObjectMapper mapper = getObjectMapper();
        MappingJackson2MessageConverter converter = getMappingJackson2MessageConverter(mapper);
        return new QueueMessagingTemplate(amazonSqs, resolver, converter);
    }

    @Bean
    @Primary
    public MappingJackson2MessageConverter getMappingJackson2MessageConverter(ObjectMapper mapper) {
        MappingJackson2MessageConverter converter = new MappingJackson2MessageConverter();
        converter.setSerializedPayloadClass(String.class);
        converter.setObjectMapper(mapper);
        return converter;
    }

    @Bean
    @Primary
    public ObjectMapper getObjectMapper() {
        ObjectMapper mapper = new ObjectMapper();
        mapper.registerModules(getJavaTimeModule());
        mapper.enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS);
        mapper.enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES);
        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
        return mapper;
    }

    @Bean
    public JavaTimeModule getJavaTimeModule() {
        return new JavaTimeModule();
    }

    @Bean
    public ResourceIdResolver getResourceIdResolver() {
        return new StackResourceRegistryDetectingResourceIdResolver();
    }

}

{code}
